src/ui/mainwindow.cpp

```diff
@@ -216,6 +216,7 @@ MainWindow::MainWindow(Application* app, SystemTrayIcon* tray_icon, OSD* osd,
   int volume = app_->player()->GetVolume();
   ui_->volume->setValue(volume);
   VolumeChanged(volume);
+  app_->player()->ReloadSettings();
```

Comment:
this is the one thing I couldn't quite get my head around while coding it. I hadn't used `QSettings` before so I tried to mimic `MainWindow` settings behavior. I agree that moving to Player's constructor would make more sense. `MainWindow` constructor also calls a series of `ReloadSettings` (including its own) so its components can init their user defined settings. 

Would it be ok moving to `Player` constructor?